firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java

```diff
@@ -418,14 +422,15 @@ public Task<Void> setIndexConfiguration(@NonNull String json) {
   @Nullable
   public PersistentCacheIndexManager getPersistentCacheIndexManager() {
     ensureClientConfigured();
-    if (persistentCacheIndexManager != null) {
-      return persistentCacheIndexManager;
-    }
+    Assert.hardAssertNonNull(
+        persistentCacheIndexManager, "persistentCacheIndexManager has not been initialized.");
+
     if (settings.isPersistenceEnabled()
         || settings.getCacheSettings() instanceof PersistentCacheSettings) {
-      persistentCacheIndexManager = new PersistentCacheIndexManager(client);
+      persistentCacheIndexManager.compareAndSet(null, new PersistentCacheIndexManager(client));
```

Comment:
Thanks for your suggestion! I am going to do
```
if (persistentCacheIndexManager.get() == null) {
  persistentCacheIndexManager.set(new PersistentCacheIndexManager(client));
}
```

Reply:
That is a race condition, because another thread could have set it to non-null after you check if it's null. You need to do the compareAndSet instead of a blind set.